src/com/jwetherell/algorithms/strings/StringFunctions.java

```diff
@@ -21,6 +27,12 @@ public static final String reverseWithStringConcat(String string) {
     }
 
     public static final String reverseWithStringBuilder(String string) {
+        if (string == null) {
+            return "";
+        }
+        if (string.isEmpty()) {
```

Comment:
Check for if a string that exists is empty, returning an empty string if it is empty.